// [START declaration]
syntax = "proto3";
package aiconnector;
// [END declaration]

// [START messages]
message Move {

    message Piece {
        string type = 1;
        int32  location = 2;
    }

    Piece newpiece = 1;
    repeated Piece removedpieces = 2;
}

message MoveMessage {

    enum RequestType {
        GET_MOVE = 0;
        SET_MOVE = 1;
    }

    enum ResponseType {
        MOVE  = 0;
        OK    = 1;
        ERROR = 2;
    }

    RequestType  requesttype = 1;
    ResponseType responsetype = 2;
    Move         move = 3;
}
// [END messages]
